parser/prism/Translator.cc

```diff
@@ -793,7 +797,37 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 }
             }
 
-            return make_unique<parser::Integer>(location, move(valueString));
+            int64_t val;
+
+            // complemented literals
+            bool hasTilde = valueString.find("~") != string::npos;
```

Comment:
We effectively already check for a tilde in the `PM_CALL_NODE` logic on line 328. I wonder if we should should move all the tilde-desugaring logic to there, and leave `PM_INTEGER_NODE` to only handle common case.

That way we centralize the weird case into `PM_CALL_NODE`, and don't have to check for `~` in two places